DatadogCore/Sources/Datadog+objc.swift

```diff
@@ -103,6 +103,52 @@ public final class objc_Datadog: NSObject {
         Datadog.clearAllData()
     }
 
+    // MARK: - User and Account Info Access
```

Comment:
suggestion/ If this is intended for KMP only, maybe we can expose this from a dedicated KMP object only in obj-c. Something like:
```swift
@objc(KMPDatadogExtension)
@objcMembers
@_spi(objc)
public final class kmp_DatadogExtension: NSObject {
    public static func subscribeToUserId(_ receiver: @escaping (String?) -> Void) {
        let core = CoreRegistry.default as? DatadogCore
        return core?.currentUserId()
    }
}
```